py/server/deephaven/stream/kafka/consumer.py

```diff
@@ -250,14 +252,26 @@ def avro_spec(
         if mapping is not None:
             mapping = _dict_to_j_func(mapping, mapped_only)
 
-        if mapping:
-            return KeyValueSpec(
-                j_spec=_JKafkaTools_Consume.avroSpec(schema, schema_version, mapping)
-            )
+        if schema.strip().startswith("{"):
+            jschema = _JKafkaTools.getAvroSchema(schema);
```

Comment:
could use json.load() to check if the supplied string is valid JSON, but delegating the check to Java is OK here.